.upstream-tests/test/cuda/proclaim_return_type.pass.cpp

```diff
@@ -60,32 +62,42 @@ struct d_callable
 
 int main(int argc, char ** argv)
 {
-#ifdef __CUDA_ARCH__
-#define TEST_SPECIFIER(...)                                                    \
-  {                                                                            \
-    test_proclaim_return_type<double>([] __VA_ARGS__ () { return 42.0; },      \
-                                      42.0);                                   \
-    test_proclaim_return_type<int>([] __VA_ARGS__ (int v) { return v * 2; },   \
-                                   42, 21);                                    \
-                                                                               \
-    int v = 42;                                                                \
-    int* vp = &v;                                                              \
-    test_proclaim_return_type<int&>(                                           \
-        [vp] __VA_ARGS__ () -> int& { return *vp; }, v);                       \
-  }
+  int v = 42;
+  int* vp = &v;
+
+  NV_IF_ELSE_TARGET(NV_IS_DEVICE, (
+    test_proclaim_return_type<int>(hd_callable{}, 42);
+    test_proclaim_return_type<int>(d_callable{}, 42);
+  ),(
+    test_proclaim_return_type<double>([]   { return 42.0; }, 42.0);
+    test_proclaim_return_type<int>   ([]   (const int v) { return v * 2; }, 42, 21);
+    test_proclaim_return_type<int&>  ([vp] () -> int& { return *vp; }, v);
+
+    test_proclaim_return_type<int>(hd_callable{}, 42);
+  ))
```

Comment:
```suggestion
  test_proclaim_return_type<int>(hd_callable{}, 42);
  test_proclaim_return_type<double>([]{ return 42.0; }, 42.0);
  test_proclaim_return_type<int>([](const int v) { return v * 2; }, 42, 21);
  test_proclaim_return_type<int&>([vp] () -> int& { return *vp; }, v);

  NV_IF_TARGET(NV_IS_DEVICE, (
    test_proclaim_return_type<int>(d_callable{}, 42);
  ))
```

Reply:
I have added the h_callable in the else branch here